Document version history (#72)
statelyai/docs: docs/versions.mdx

<!-- TODO: ![A version previewed in the Editor with an arrow pointing to the Restore button.]() -->

## Create a new machine from a version

You can also create a new machine from a version in your version history.

1. Open the **Version history** panel from the right tool menu.
2. Select the version from which you want to create a new machine. The version will be previewed in the Editor.
3. Use the copy icon button in the top bar to create a new machine from that version.

<!-- TODO: ![A version previewed in the Editor with an arrow pointing to the Copy button.]() -->

Your new machine will be created and opened in the same project with the same name appended with “(copy).”

## Delete a version

1. Open the **Version history** panel from the right tool menu.
2. Select the version you want to delete. The version will be previewed in the Editor.
3. Use the delete icon button in the top bar to delete that version.
4. The current version will then be opened in the Editor.

<!-- TODO: ![A version previewed in the Editor with an arrow pointing to the Delete button.]() -->

## Team roles and versions

> Team viewers can only create new versions of machines in their team’s projects. Team owners, admins, and editors can view, create, restore, and delete versions of a team project’s machine.

Reviewer: ```suggestion
All team members can view past versions of machines in their team’s projects. Team owners, admins, and editors can also create and restore versions. Only owners and admins can delete versions.
```

| Capability                               | Owner  | Admin  | Editor | Viewer |
| ---------------------------------------- | ------ | ------ | ------ | ------ |
| View team machine versions               | ✅ Yes | ✅ Yes | ✅ Yes | ✅ Yes |
| Create new versions of team machines     | ✅ Yes | ✅ Yes | ✅ Yes | ✅ Yes |
| Restore versions of team machines        | ✅ Yes | ✅ Yes | ✅ Yes | 🚫 No  |

> | Delete versions of team machines         | ✅ Yes | ✅ Yes | ✅ Yes | 🚫 No  |

Reviewer: ```suggestion
| Delete versions of team machines         | ✅ Yes | ✅ Yes | 🚫 No | 🚫 No  |
```

### Initial versions

An initial version is automatically made for new machines created when:

- you import a machine from code
- you fork a machine from an existing machine or project

> - you duplicate an existing machine or project

Reviewer: Might be more accurate to say "you duplicate an existing machine in its project" 

I don't think we "duplicate" projects yet although that could be cool, maybe.

### Initial versions

> An initial version is created when a new machine is created or imported. Autosaved machines are is “Initial version” and have a Stately Bot icon.

Author: @kevinmaes I used your copy but changed it into a list for readability.

> - you import a machine from code

Reviewer: @laurakalbag We need to remove this one for now. We had trouble with the layout so we don't create an initial version anymore for import from code.

Author: @kevinmaes done!
